File: EasySearch.FrontEnd.Web/src/app/chatbot-system/models/dtos/chatGptDto.ts
export class GeoLocation {
  public latitude: number;
  public longitude: number;

  constructor(latitude: number, longitude: number) {
    this.latitude = latitude;
    this.longitude = longitude;
  }
}

export class ChatGptDto {
  text: string = '';
  geoLocation: GeoLocation | null = null;

  constructor(text: string, geoLocation: GeoLocation | null) {
    this.text = text;
    this.geoLocation = geoLocation;
  }
}

File: EasySearch.MicroServices.ChatbotApi/Models/ChatGpt/ChatGptDto.cs
namespace EasySearch.MicroServices.ChatbotApi.Models.ChatGpt
{

    public class GeoLocation
    {
        public double Latitude { get; set; }
        public double Longitude { get; set; }
    }

    public class ChatGptDto
    {
        public string Text { get; set; } = "";
        public GeoLocation? GeoLocation { get; set; } = null;
    }

}
